controllers/core/openstackcontrolplane_controller.go

```diff
@@ -159,10 +159,14 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneKeystoneAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneKeystoneAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlanePlacementAPIReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlanePlacementAPIReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneGlanceReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneGlanceReadyInitMessage),
-			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneCinderReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneCinderReadyInitMessage),
 			condition.UnknownCondition(corev1beta1.OpenStackControlPlaneNovaReadyCondition, condition.InitReason, corev1beta1.OpenStackControlPlaneNovaReadyInitMessage),
 		)
 
+		// Initialize optional service conditions if enabled
+		if instance.Spec.Cinder.Enabled {
```

Comment:
I've seen a few users not using even nova (just for keystone and swift, for example). Also, some users would want to disable ovs/ovn when they use 3rd party SDN.

So I'm wondering whether it makes sense to make sure all services can be disabled (with some dependency validation as webhook ?)

Reply:
Do we anticipate a minimum number of services deployed to constitute a viable, supportable OSP deployment? If so, maybe we can just group core and optional services for now. Until we have a better system for managing this.

I would ultimately like to stop handling this on a service-by-service basis and implement something a little more dynamic. But that is still a work in progress for me while I figure out the best implementation to propose.

Reply:
> Do we anticipate a minimum number of services deployed to constitute a viable, supportable OSP deployment?

In case we support a use case with only swift and keystone then probably the core would be only keystone, mariadb and memcached ?
I'd rather avoid defining "the must items" strictly and leave the selections to users. That would be more consistent with what we expose now (all services have enabled/disabled flag)

Reply:
I agree, we should probably have the same for all the services as they have enabled/disabled parameter

Reply:
Ok, no worries. Let's do it for all services.